events/sink.go

```diff
@@ -56,3 +64,49 @@ func (s *eventSink) Write(evt goevents.Event) error {
 func (s *eventSink) Close() error {
 	return nil
 }
+
+func adaptEvent(topic string, evt Event) filters.Adaptor {
+	return filters.AdapterFunc(func(fieldpath []string) (string, bool) {
+		if len(fieldpath) == 0 {
+			return "", false
+		}
+
+		s := evt.(*sinkEvent)
+
+		switch fieldpath[0] {
```

Comment:
I started it [here](df408cd020dc4f5a0580ff82619f0f63f286405c) to see how it looks.  wdyt?